common/configurable/src/main/java/io/helidon/common/configurable/WrappedVirtualExecutor.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.configurable;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+class WrappedVirtualExecutor implements ExecutorService {
```

Comment:
fundamentally, why do we need executors in a "common configurable" module?  Seems like this should be a higher level function...  Needs javadoc so maybe I'd understand it better after it was written.

Reply:
This class does not need javadoc, as it is not generated for it (it is a package local class).
This is a refactoring of original approach that used reflection. The new approach is reflection free.
Common configurable module was created for classes that are used by multiple other modules that can be configured through config. Executor services are used for example by Fault tolerance, MP. 
Theoretically we could extract executor services into a new module, such as `common-executors` if we wanted to